cli/cppcheckexecutor.cpp

```diff
@@ -62,8 +62,7 @@
 #   include <ucontext.h>
 #endif
 #ifdef __linux__
-#include <sys/syscall.h>
-#include <sys/types.h>
+#include <syscall.h>
```

Comment:
This change is incorrect.  I've been improving the man-pages so that the correct includes are more clear; see membarrier(2) for example:

```
SYNOPSIS
       #include <linux/membarrier.h> /* Definition of MEMBARRIER_* constants */
       #include <sys/syscall.h>      /* Definition of SYS_* constants */
       #include <unistd.h>

       int syscall(SYS_membarrier, int cmd, unsigned int flags, int cpu_id);

       Note: glibc provides no wrapper for membarrier(), necessitating the use
       of syscall(2).
```

I opened an issue in iwyu(1) regarding that:

https://github.com/include-what-you-use/include-what-you-use/issues/1036

Reply:
@alejandro-colomar 
Thanks for pointing that out. I implicitly addressed this in the refactoring that is https://github.com/danmar/cppcheck/pull/4359.